src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -238,6 +239,26 @@ namespace Utils
             return END_ITERATOR;
         }
 
+        /**
+         * @brief Compacts the key range in the RocksDB database.
+         *
+         * This function triggers compaction for the entire key range in the RocksDB
+         * database. Compaction helps to reduce the storage space used by the database
+         * and improve its performance by eliminating unnecessary data.
+         *
+         * @note This function uses default compact range options.
+         *
+         * @see rocksdb::CompactRangeOptions
+         */
+        void compactDatabase()
+        {
+            // Create compact range options with default settings
+            rocksdb::CompactRangeOptions compactOptions;
+
+            // Perform compaction for the entire key range
+            m_db->CompactRange(compactOptions, nullptr, nullptr);
```

Comment:
This seems to be an advanced feature that we should fully understand before including it
https://github.com/facebook/rocksdb/wiki/Manual-Compaction

It doesn't seem to be related to the compression, it's more like  a "vacuum" .
But if we are going to use it anyway, we shouldn't run it after every message processed. It could be only at startup for example.

Reply:
I believe that the concept of compacting the CVE database is well-defined. This particular database is not likely to be used and aligns with the second point on the list: "Force the data to go through the compaction filter in order to consolidate it."

Also, we call the compaction once all messages have been processed.

Reply:
Ok, thank you.